### ISSUE TITLE
Remove useless Layout/SpaceAfterColon Cop

### ISSUE THREAD

**Issue by AdrienSldy** [rubocop-hq/rubocop#7157]
## Is your feature request related to a problem? Please describe.

The `Layout/SpaceAfterColon`, checks space needed between colon and value in ruby 1.9 hash syntax. It's already check with Layout/AlignHash.
Each style describe in this latter cop guarantee at least one space between separators and values.

Also since #6649 release in [0.70.0 (2019-05-21)](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md#0700-2019-05-21), user can choose multiple alignement style, reinforcing the feeling than the `Layout/SpaceAfterColon` cop is become useless.

## Describe the solution you'd like

Remove this cop which has become useless.

-----------

**What do you think ?**

**Comment by koic** [comment]
I think that roles of `Layout/SpaceAfterColon` cop and `Layout/AlignHash` cop are different.

`Layout/AlignHash` cop is a cop for Hash style. For example, `Layout/SpaceAfterColon` cop has detected a kwargs. This is not detected by `Layout/AlignHash` cop.

```console
% cat example.rb
# frozen_string_literal: true

def m(foo:1, bar:2)
end

% rubocop
Inspecting 1 file
C

Offenses:

example.rb:3:1: C: Style/EmptyMethod: Put empty method definitions on a single line.
def m(foo:1, bar:2) ...
^^^^^^^^^^^^^^^^^^^
example.rb:3:10: C: Layout/SpaceAfterColon: Space missing after colon.
def m(foo:1, bar:2)
         ^
example.rb:3:17: C: Layout/SpaceAfterColon: Space missing after colon.
def m(foo:1, bar:2)
                ^

1 file inspected, 3 offenses detected
```

It can not be detected when `Layout/SpaceAfterColon` is disabled.

```console
% cat .rubocop.yml
Layout/SpaceAfterColon:
  Enabled: false

% rubocop
Inspecting 1 file
C

Offenses:

example.rb:3:1: C: Style/EmptyMethod: Put empty method definitions on a single line.
def m(foo:1, bar:2) ...
^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

So `Layout/SpaceAfterColon` cop cannot be removed as it works.

**Comment by AdrienSldy** [comment]
@koic

You agree, I forgot to check the inline hash.

**Comment by EDISONchen00** [comment]
N.
